netsuite/10.0/modules/ROOT/pages/netsuite-call-restlets-example.adoc

```diff
@@ -0,0 +1,159 @@
+= Call NetSuite RESTlets Example
+
+This example shows how to call NetSuite RESTlets that are deployed in NetSuite instances to:
+
+* Create a customer record
+* Retrieve a customer record
+* Delete a record
+
+== Prerequisites
+
+To follow these examples, you must have:
+
+* Java 8
+* Anypoint Studio 7.0.x
+* Mule Runtime 4.1.1 or later
+* DataWeave 2.0
+* Access to a NetSuite instance with credentials in the `mule-app.properties` file
+
+[[deploy-a-script-as-restlet]]
+== Deploy a Script as a RESTlet
+
+. Create a sample JavaScript script:
++
+[source,javascript,linenums]
+----
+// Get a standard NetSuite record
+function getRecord(datain)
+{
+    return nlapiLoadRecord(datain.recordtype, datain.id);
+    // for example, recordtype="customer", id="769"
+}
+
+// Create a standard NetSuite record
+
+function createRecord(datain)
+{
+    var err = new Object();
+
+    // Validate if mandatory record type is set in the request
+    if (!datain.recordtype)
+    {
+        err.status = "failed";
+        err.message= "missing recordtype";
+        return err;
+    }
+
+    var record = nlapiCreateRecord(datain.recordtype);
+
+    for (var fieldname in datain)
+    {
+      if (datain.hasOwnProperty(fieldname))
+      {
+        if (fieldname != 'recordtype' && fieldname != 'id')
+        {
+          var value = datain[fieldname];
+            if (value && typeof value != 'object')
+            // ignore other type of parameters
+            {
+              record.setFieldValue(fieldname, value);
+            }
+        }
+      }
+    }
+    var recordId = nlapiSubmitRecord(record);
+    nlapiLogExecution('DEBUG','id='+recordId);
+    var nlobj = nlapiLoadRecord(datain.recordtype,recordId);
+    return nlobj;
+}
+
+// Delete a standard NetSuite record
+function deleteRecord(datain)
+{
+  nlapiDeleteRecord(datain.recordtype, datain.id);
+  // for example: recordtype="customer", id="769"
+
+}
+----
++
+. Enable SuiteScript and NetSuite Web Services in your account. 
+.. Log in to NetSuite.
+.. Click *Set Up > Company > Enable Features > SuiteCloud*.
++
+image::netsuite-enable-suitescript.png[Enable SuiteScript]
++
+. Create a new script and upload the file you <<deploy-a-script-as-restlet,previously created>>.
+.. Go to *Customization > Scripting > Scripts > New*.
+.. Select the script file, click *Create Script Record*, and select *RESTlet*.
++
+image::netsuite-script-type.png[Select Script Type]
++
+. Fill out the form using the content of the script you uploaded, and deploy the script.
++
+image::netsuite-setup-script.png[Setup Script]
++
+. After selecting your audience you see the following page:
++
+image::netsuite-script-deployment.png[Script Deployment]
++ 
+Take note of the script and deploy numbers in the external URL as you will need them later to call the RESTlet.
+
+== Import the Project
```

Comment:
Import what project? 

How does this relate to the previous task of **deploying a script** as a RESTlet, and how do they together exemplify **calling NetSuite RESTlets**?

Again, the container task is calling RESTlets, and the intro text says that you **call them so that you can deploy them to create, retrieve, and delete records**. But the subtasks are **deploying** a script as a RESTlet and then **importing** a project.

The verbs throughout the set are not consistent. What *is* the user task here?